pkg/controller.v1alpha3/experiment/experiment_util.go

```diff
@@ -109,3 +110,25 @@ func (r *ReconcileExperiment) updateFinalizers(instance *experimentsv1alpha3.Exp
 		return reconcile.Result{Requeue: true}, err
 	}
 }
+
+func (r *ReconcileExperiment) shutdownSuggestionServer(instance *experimentsv1alpha3.Experiment) error {
+	if instance.Spec.ResumeExperiment {
+		return nil
+	}
+	suggestionToDelete := &suggestionsv1alpha3.Suggestion{}
+	err := r.Get(context.TODO(), types.NamespacedName{
+		Namespace: instance.Namespace,
+		Name:      instance.Name,
+	}, suggestionToDelete)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	err = r.Delete(context.TODO(), suggestionToDelete)
```

Comment:
I believe here you will delete Suggestion CR not just Suggestion deployment.
Is that correct?
Does user want to check Suggestion CR even if Suggestion Deployment is deleted?